cmake/modules/AddSwift.cmake

```diff
@@ -1025,14 +1090,17 @@ function(_add_swift_library_single target name)
     RESULT_VAR_NAME link_flags
       )
 
-  if(SWIFT_ENABLE_GOLD_LINKER AND
-     "${SWIFT_SDK_${SWIFTLIB_SINGLE_SDK}_OBJECT_FORMAT}" STREQUAL "ELF")
-    list(APPEND link_flags "-fuse-ld=gold")
-  endif()
-  if(SWIFT_ENABLE_LLD_LINKER OR
-     ("${SWIFTLIB_SINGLE_SDK}" STREQUAL "WINDOWS" AND
-      NOT "${CMAKE_SYSTEM_NAME}" STREQUAL "WINDOWS"))
-    list(APPEND link_flags "-fuse-ld=lld")
+  # MSVC doesn't recognise -fuse-ld
+  if (NOT "${CMAKE_C_COMPILER_ID}" STREQUAL "MSVC")
+    if(SWIFT_ENABLE_GOLD_LINKER AND
+       "${SWIFT_SDK_${SWIFTLIB_SINGLE_SDK}_OBJECT_FORMAT}" STREQUAL "ELF")
+      list(APPEND link_flags "-fuse-ld=gold")
+    endif()
+    if(SWIFT_ENABLE_LLD_LINKER OR
+       ("${SWIFTLIB_SINGLE_SDK}" STREQUAL "WINDOWS" AND
+        NOT "${CMAKE_SYSTEM_NAME}" STREQUAL "WINDOWS"))
+      list(APPEND link_flags "-fuse-ld=lld")
+    endif()
```

Comment:
How would MSVC understand `-fuse-ld=lld` even if we're cross-compiling to Linux?

I've changed this code BTW to:
```
  # MSVC doesn't recognise -fuse-ld.
  if ("${CMAKE_C_COMPILER_ID}" STREQUAL "Clang")
    if(SWIFT_ENABLE_GOLD_LINKER AND
       "${SWIFT_SDK_${SWIFTLIB_SINGLE_SDK}_OBJECT_FORMAT}" STREQUAL "ELF")
      list(APPEND link_flags "-fuse-ld=gold")
    endif()
    if(SWIFT_ENABLE_LLD_LINKER OR
       ("${SWIFTLIB_SINGLE_SDK}" STREQUAL "WINDOWS" AND
        NOT "${CMAKE_SYSTEM_NAME}" STREQUAL "WINDOWS"))
      list(APPEND link_flags "-fuse-ld=lld")
    endif()
  endif()
```
Perhaps this addresses the cross-compilation concerns?